docs/DeveloperGuide.md

```diff
@@ -1,29 +1,136 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started](#setting-up-getting-started)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#UI-Component)
+    * [Parser Component](#parser-component)
+    * [BankWithUs Component](#bankwithus-component)
+    * [Account Component](#account-component)
+    * [AccountList Component](#accountlist-component)
+    * [SaveGoal Component](#saveGoal-component)
+    * [Storage Component](#storage-component)
+    * [Withdrawal Limit Component](#withdraw limit checker-Component)
+    * [Transaction Component](#transaction-Component)
+    * [TransactionList Component](#transactionlist-component)
+* [Implementation](#implementation)
+    * [Account related Feature](#Account-feature)
+    * [saveGoal related Feature](#SaveGoal-feature)
+    * [transaction Related Feature](#transaction-feature)
+
+* [Appendix: Requirements](#appendix-requirements)
+    * [Product scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+    * [User Stories](#user-stories)
+    * [Non-Functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+
+---
+
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```

Comment:
There are many placeholder sentences left in the code. I think it would be better to remove them.